robot.py

```diff
@@ -111,20 +106,33 @@ def teleopPeriodic(self) -> None:
                 self.chassis.snap_to_heading(-math.radians(dpad))
 
         # Set current robot direction to forward
-        if self.gamepad.getXButton():
+        if self.gamepad.getBButton():
             self.chassis.reset_yaw()
 
-        # stop rumble after time
-        if self.rumble_timer.hasElapsed(self.rumble_duration):
-            self.gamepad.setRumble(wpilib.XboxController.RumbleType.kBothRumble, 0)
+        # Reset Odometry
+        if self.gamepad.getStartButtonPressed():
+            self.chassis.reset_odometry()
```

Comment:
One of these simply reads the button state, whereas the other reads the rising edge of the button. Should these be consistent?